samples/media/ultrahdr/src/main/java/com/example/platform/media/ultrahdr/common/ColorModeControls.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.platform.media.ultrahdr.common
+
+import android.content.Context
+import android.content.pm.ActivityInfo
+import android.os.Build
+import android.util.AttributeSet
+import android.util.Log
+import android.view.Display
+import android.view.LayoutInflater
+import android.view.Window
+import android.widget.LinearLayout
+import androidx.annotation.RequiresApi
+import com.example.platform.media.ultrahdr.R
+import com.example.platform.media.ultrahdr.databinding.ColorModeControlsBinding
+import java.util.function.Consumer
+
+/**
+ * This class creates a custom view that is able to change the Activity's Color Mode to HDR, which
+ * enable UltraHDR Viewing. The [setWindow] method must be called in order for the UltraHDR viewing
+ * to happen.
+ */
+@RequiresApi(Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
+class ColorModeControls : LinearLayout, WindowObserver {
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
+
+    /**
+     *  Android ViewBinding.
+     */
+    private var _binding: ColorModeControlsBinding? = null
+    private val binding get() = _binding!!
+
+    /**
+     * Reference to [Window]. This should come from the currently active activity.
+     */
+    private var window: Window? = null
+
+    init { // Inflate binding
+        _binding = ColorModeControlsBinding.inflate(LayoutInflater.from(context), this, true)
+    }
+
+    private val hdrSdrRatioListener = Consumer<Display> { display ->
+        Log.d(TAG, "HDR/SDR Ratio Changed ${display.hdrSdrRatio}")
+        post { updateModeInfoDisplay() }
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+        display?.run {
+            registerHdrSdrRatioChangedListener({ it.run() }, hdrSdrRatioListener)
+        }
+    }
+
+    override fun onDetachedFromWindow() {
```

Comment:
Question: this happens when the activity is destroyed or the view is removed from the viewgroup, but if the app goes onPause this won't happen. Is it okay to still have the listener registered and HDR mode on? Or is it best practice to turn on/off onresume/onpause?

Reply:
Also, you are not updating the mode on detach. Shouldn't you do it? Otherwise the activity will be kept as HDR even if the view is not there anymore

Reply:
Ah yes, makes sense